opendevin/action/fileop.py

```diff
@@ -19,13 +24,28 @@ def resolve_path(file_path):
 
 @dataclass
 class FileReadAction(ExecutableAction):
+    """
+    Reads a file from a given path up to 100 lines.
+    Default lines 0:100
+    """
     path: str
+    start_index: int = 0
+    thoughts: str = ''
     action: str = ActionType.READ
 
     def run(self, controller) -> FileReadObservation:
         path = resolve_path(self.path)
         with open(path, 'r', encoding='utf-8') as file:
-            return FileReadObservation(path=path, content=file.read())
+            all_lines = file.readlines()
+            total_lines = len(all_lines)
+            if total_lines >= 100:
+                end_index = self.start_index + 100 if total_lines - \
+                    self.start_index - 100 >= 0 else -1
+                code_slice = all_lines[self.start_index: end_index]
+            else:
+                code_slice = all_lines[:]
+            code_view = ''.join(code_slice)
```

Comment:
maybe we pass in `total_lines` as a parameter?

Reply:
I like this idea!